courseraresearchexports/exports/utils.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Coursera
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from courseraresearchexports.exports.constants import COURSE_API, PARTNER_API,\
+    EXPORT_TYPE_EVENTING, EXPORT_TYPE_SCHEMAS
+from urlparse import urlparse
+from tqdm import tqdm
+import requests
+import os
+import logging
+import zipfile
+
+
+def extract_export_archive(export_archive, dest, delete_archive=True):
+    """
+    Extracts a compressed folder
+    :param export_archive:
+    :param dest:
+    :param delete_archive: delete the archive after extracting
+    :return dest:
+    """
+    try:
+        with zipfile.ZipFile(export_archive, 'r') as z:
+            z.extractall(dest)
+        if delete_archive:
+            os.remove(export_archive)
+        return dest
+    except:
+        logging.error('Error in extracting export archive {} to {}'.format(
+            export_archive, dest))
+        raise
+
+
+def download_export(export_job, dest):
+    """
+    Download an export archive if ready given an export_job_id
+    :param export_job:
+    :param dest:
+    :return output_filename:
+    """
+    if export_job['exportType'] == EXPORT_TYPE_EVENTING:
+        logging.error('Generate download links to download eventing exports')
+        raise ValueError('Require {} job'.format(EXPORT_TYPE_SCHEMAS))
+    elif 'downloadLink' not in export_job:
```

Comment:
Let's make the error handling more helpful to the end user (e.g for eventing export, we can advertise the command needed to generate links, for export job, we can dump the status of the job)